internal/repository/repository.go

```diff
@@ -0,0 +1,26 @@
+package repository
+
+import (
+	"sublinks/sublinks-federation/internal/db"
+)
+
+type Repository interface {
+	Save(interface{}) error
+	Find(*interface{}, ...interface{}) error
+}
+
+type RepositoryImpl struct {
```

Comment:
I ended up going with a generic implementation as the only thing that was different in each repository was the the type reference. This just led to a bunch of boilerplate copy-paste code that made my skin crawl 😱